source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -353,6 +353,22 @@ void JsonTranscoderFilter::setDecoderFilterCallbacks(
   decoder_callbacks_ = &callbacks;
 }
 
+void JsonTranscoderFilter::encodeGrpcStatusMessage(Http::HeaderMap& trailers) {
+  const absl::optional<Grpc::Status::GrpcStatus> grpc_status =
+      Grpc::Common::getGrpcStatus(trailers);
+  if (!grpc_status || grpc_status.value() == Grpc::Status::GrpcStatus::InvalidCode) {
+    response_headers_->Status()->value(enumToInt(Http::Code::ServiceUnavailable));
```

Comment:
if upstream doesn't return grpc-status, it usually means the HTTP status is not 200 (5xx etc), in that case shall we override the HTTP status?

Reply:
Updated, not to set Status() if upstream grpc-status is not available.

Reply:
`if (grpc_status)` should be checked before `if (grpc_status.value() == ...)`

Reply:
Done.  Thanks for catching this crash.